jdbc-adapter/src/main/java/org/polypheny/db/adapter/jdbc/stores/MonetdbStore.java

```diff
@@ -45,24 +53,65 @@
+@AdapterSettingString(name = "database", defaultValue = "polypheny", description = "JDBC port number on the remote MonetDB instance.", position = 3, appliesTo = DeploySetting.REMOTE)
+@AdapterSettingString(name = "username", defaultValue = "polypheny", description = "Name of the database to connect to.", position = 4, appliesTo = DeploySetting.REMOTE)
+@AdapterSettingString(name = "password", defaultValue = "polypheny", description = "Username to be used for authenticating at the remote instance.")
+@AdapterSettingInteger(name = "maxConnections", defaultValue = 25, description = "Password to be used for authenticating at the remote instance.")
 public class MonetdbStore extends AbstractJdbcStore {
 
-    public static final String ADAPTER_NAME = "MonetDB";
+    private String host;
+    private String database;
+    private String username;
 
-    public static final String DESCRIPTION = "MonetDB is an open-source column-oriented database management system. It is based on an optimistic concurrency control.";
 
-    public static final List<AdapterSetting> AVAILABLE_SETTINGS = ImmutableList.of(
-            new AdapterSettingString( "host", false, true, false, "localhost" ),
-            new AdapterSettingInteger( "port", false, true, false, 50000 ),
-            new AdapterSettingString( "database", false, true, false, "polypheny" ),
-            new AdapterSettingString( "username", false, true, false, "polypheny" ),
-            new AdapterSettingString( "password", false, true, false, "" ),
-            new AdapterSettingInteger( "maxConnections", false, true, false, 25 )
-    );
+    public MonetdbStore( int storeId, String uniqueName, final Map<String, String> settings ) {
+        super( storeId, uniqueName, settings, MonetdbSqlDialect.DEFAULT, true );
+    }
 
 
-    public MonetdbStore( int storeId, String uniqueName, final Map<String, String> settings ) {
-        super( storeId, uniqueName, settings, createConnectionFactory( settings, MonetdbSqlDialect.DEFAULT ), MonetdbSqlDialect.DEFAULT, true );
+    @Override
+    protected ConnectionFactory deployDocker( int dockerInstanceId ) {
+        DockerManager.Container container = new ContainerBuilder( getAdapterId(), "topaztechnology/monetdb:11.37.11", getUniqueName(), dockerInstanceId )
+                .withMappedPort( 50000, Integer.parseInt( settings.get( "port" ) ) )
+                .withEnvironmentVariables( Arrays.asList( "MONETDB_PASSWORD=" + settings.get( "password" ), "MONET_DATABASE=monetdb" ) )
+                .build();
+        DockerManager.getInstance().initialize( container ).start();
+
+        host = container.getHost();
+        database = "monetdb";
+        username = "monetdb";
+
+        // TODO: Wait until ready
+        try {
+            TimeUnit.SECONDS.sleep( 15 );
+        } catch ( InterruptedException e ) {
+            // ignore
+        }
```

Comment:
We need a better solution for this.

Reply:
I implemented https://github.com/polypheny/Polypheny-DB/blob/201f51223ef177570fb4834866fbd3a1b14b9679/jdbc-adapter/src/main/java/org/polypheny/db/adapter/jdbc/stores/MonetdbStore.java#L83 
and https://github.com/polypheny/Polypheny-DB/blob/201f51223ef177570fb4834866fbd3a1b14b9679/jdbc-adapter/src/main/java/org/polypheny/db/adapter/jdbc/stores/MonetdbStore.java#L320-L346
in commit https://github.com/polypheny/Polypheny-DB/pull/314/commits/201f51223ef177570fb4834866fbd3a1b14b9679#diff-dc963a8404911e3b9dc858d2fffdc75ac38fa5c555649cdd8d7838740531cd6c

for MonetDB, PostgresDB and CassandraDB, which removes this.